attentionbench/attention_utils.py

```diff
@@ -0,0 +1,124 @@
+from utils import *
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+
+@dataclass
+class AttentionConfig:
+    B: int
+    M: int
+    N: int
+    K1: int
+    K2: int
+    dtype: str
+
+    def get_name(self) -> str:
+        return f"attention_{self.B}x{self.M}x{self.N}x{self.K1}x{self.K2}x{self.dtype}"
+
+    def get_query_shape(self) -> str:
+        return f"{self.B}x{self.M}x{self.K1}x{self.dtype}"
+
+    def get_key_shape(self) -> str:
+        return f"{self.B}x{self.K2}x{self.K1}x{self.dtype}"
+
+    def get_value_shape(self) -> str:
+        return f"{self.B}x{self.K2}x{self.N}x{self.dtype}"
+
+    def get_output_shape(self) -> str:
+        return f"{self.B}x{self.M}x{self.N}x{self.dtype}"
+
+    def get_byte_count(self) -> int:
+        if "bf" in self.dtype:
+            bytes_per_element = int(self.dtype[2:]) // 8
+        else:
+            bytes_per_element = int(self.dtype[1:]) // 8
+        element_count = (
+            (self.B * self.M * self.K1)
+            + (self.B * self.K2 * self.K1)
+            + (self.B * self.K2 * self.N)
```

Comment:
Usually for the element count for data movement we also account for the output size, so BxMxN. Any reason we are leaving out here?

Reply:
I'm not too sure what's the right answer here, so I just added the output size also.